build-tools/packages/build-cli/src/commands/generate/releaseNotes.ts

```diff
@@ -0,0 +1,202 @@
+					return input;
+				}
+
+				throw new Error(`Invalid release type: ${input}`);
+			},
+		})(),
+		out: Flags.file({
+			description: `Output the results to this file.`,
+			required: true,
+			default: "RELEASE_NOTES.md",
+			// exists: true
+		}),
+		...BaseCommand.flags,
+	} as const;
+
+	static readonly examples = [
+		{
+			description: `Generate release notes for a minor release of the client release group.`,
+			command: "<%= config.bin %> <%= command.id %> -g client -t minor",
+		},
+		{
+			description: `You can output a different file using the --out flag.`,
+			command:
+				"<%= config.bin %> <%= command.id %> -g client -t minor --out RELEASE_NOTES/2.1.0.md",
+		},
+	];
+
+	public async run(): Promise<string> {
+		const context = await this.getContext();
+		const { flags, logger } = this;
+
+		const releaseGroup = context.repo.releaseGroups.get(flags.releaseGroup);
+		if (releaseGroup === undefined) {
+			this.errorLog(`Unknown release group: ${flags.releaseGroup}`);
+			this.exit(2);
+		}
+
+		const { releaseNotes: releaseNotesConfig } = loadFluidBuildConfig(
+			context.gitRepo.resolvedRoot,
+		);
+		if (releaseNotesConfig === undefined) {
+			this.errorLog(
+				`No releaseNotes config found. Make sure the 'releaseNotes' section of the build config exists.`,
+			);
+			this.exit(2);
+		}
+
+		const changesetDir = path.join(releaseGroup.directory, DEFAULT_CHANGESET_PATH);
+		const changesets = await loadChangesets(changesetDir, logger);
+
+		const { version } = releaseGroup;
+		const header = `<!-- THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. -->`;
+		const footer = `### 🛠️ Start Building Today!\n\nPlease continue to engage with us on GitHub
+[Discussion](https://github.com/microsoft/FluidFramework/discussions) and
+[Issue](https://github.com/microsoft/FluidFramework/issues) pages as you adopt Fluid Framework!
+`;
+		const intro = `# Fluid Framework v${version}\n\n## Contents`;
+
+		this.info(`Loaded ${changesets.length} changes.`);
+		assert(flags.releaseType !== undefined, `Release type must be provided.`);
+
+		const bySection = groupBySection(changesets);
+
+		{
+			const unknownSection = bySection.get(UNKNOWN_SECTION);
+			for (const changeset of unknownSection ?? []) {
+				if (changeset.additionalMetadata?.includeInReleaseNotes !== false) {
+					this.warning(
+						`Changeset doesn't map to known sections. Check its metadata: ${changeset.sourceFile}`,
+					);
+				}
+			}
+
+			const sectionsInChangesets = new Set<string>(bySection.keys());
+			const configuredSections = new Set<string>(
+				releaseNotesConfig.sections.map((s) => s.name),
+			);
+			const unknownSections = difference(sectionsInChangesets, configuredSections);
+			for (const section of unknownSections) {
+				if (section !== UNKNOWN_SECTION) {
+					this.warning(
+						`Could not find a configuration for a section named "${section}". The changesets in this section will be omitted.`,
+					);
+				}
```

Comment:
Not sure I love skipping them by default. If someone misses the warning in console output and assumes the generated file has everything, those changes will probably just not make it to the final notes. Would it be better to put them under a final "Unknown section (please review)" header after "Other changes", for people to review and reclassify them (or decide they can indeed be removed)?

Reply:
I'd prefer to error in this case if we really don't want to omit by default. That's a little annoying, but if the
concern is that things get omitted, erroring is the best way to ensure everything has a category.